src/proof/lfsc/lfsc_node_converter.cpp

```diff
@@ -604,11 +601,43 @@ TypeNode LfscNodeConverter::postConvertType(TypeNode tn)
 
 std::string LfscNodeConverter::getNameForUserName(const std::string& name)
 {
+  std::stringstream ssan;
   std::stringstream ss;
-  ss << "cvc." << name;
+  // We also must sanitize symbols that are not allowed in LFSC identifiers
+  // here. For the sake of generating unique symbols, we record the santization
+  // as a prefix before the "." separating the user name. Thus, e.g.
+  //   cvc.|a b|
+  // is converted to:
+  //   cvc_32@1.ab
+  // where "32" is the unicode value of " ", 32@1 indicates that space was
+  // removed from the user string at position 1.
+  std::string sname = name;
+  size_t sanCount = 0;
+  size_t found;
+  do
+  {
+    found = sname.find_first_not_of("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+                          "0123456789~!@$%^&*_-+=<>.?/");
+    if (found!=std::string::npos)
+    {
+      // erase the character, print that we eliminated that character
+      ssan << "_" << static_cast<size_t>(sname[found])+sanCount << "@" << found;
```

Comment:
Why add `sanCount` here?

Reply:
Since we are updating sname in this loop, sanCount ensures we print the position in the original string.